packages/material-ui-styles/src/index.js

```diff
@@ -1,3 +1,23 @@
+/* Warning if there are several instances of @material-ui/styles */
+if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
+  window['__@material-ui/styles-init__'] = window['__@material-ui/styles-init__'] || 0;
+
+  if (window['__@material-ui/styles-init__'] === 1) {
```

Comment:
Should probably use `global` to get these for SSR as well?

Reply:
The server case is ignored early in the branch logic: (`typeof window !== 'undefined'`). But you are correct; If we want to handle SSR too, we should use our global helper. Should we handle SSR? Let me see the sc issue history to try to understand why they don't. Maybe there is an issue with hot reloading on the server or the handling the client side covers 90% of the problem.

Reply:
Ok, let's be greedy, let's try.